packages/components/src/components/hds/app-side-nav/index.hbs

```diff
@@ -0,0 +1,37 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+{{! IMPORTANT: we need to add "squishies" here (~) because otherwise the whitespace added by Ember causes the empty element to still have visible padding - See https://handlebarsjs.com/guide/expressions.html#whitespace-control }}
+
+<Hds::AppSideNav::Base
+  class={{concat this.classNames (unless (has-block "header") " hds-app-side-nav--is-headerless")}}
+  ...attributes
+  {{on "transitionstart" (fn this.setTransition "start")}}
+  {{on "transitionend" (fn this.setTransition "end")}}
+  {{! @glint-expect-error - https://github.com/josemarluedke/ember-focus-trap/issues/86 }}
+  {{focus-trap isActive=this.shouldTrapFocus}}
+  {{did-insert this.didInsert}}
+>
+  <:root>
+    {{#if this.showToggleButton}}
+      {{! template-lint-disable no-invalid-interactive}}
+      <div class="hds-app-side-nav__overlay" {{on "click" this.toggleMinimizedStatus}} />
+      {{! template-lint-enable no-invalid-interactive}}
+      <Hds::AppSideNav::ToggleButton
+        aria-label={{this.ariaLabel}}
+        @icon={{if this.isMinimized "chevrons-right" "chevrons-left"}}
+        {{on "click" this.toggleMinimizedStatus}}
+      />
+    {{/if}}
+  </:root>
+  <:header as |Header|>
+    {{~yield (hash Header=Header isMinimized=this.isMinimized) to="header"~}}
+  </:header>
+  <:body as |Body|>
+    {{~yield (hash Body=Body isMinimized=this.isMinimized) to="body"~}}
+  </:body>
+  <:footer as |Footer|>
```

Comment:
I thought it would be more flexible/friendly for consumers if we left optional named blocks where they could insert "header" or "footer" content if they really needed/wanted to.

For the header area, I did remove two child components we had that were originally used within it but I left the `:header` named block itself.

Reply:
@alex-ju (or others reading this) Do you have an opinion on whether we should keep the `:header` & `:footer` named blocks or if it would be best to remove them?

Reply:
I don't have the complete historical context of these, but I believe the `:footer` was added for the org selector (that was custom across products at the time), likewise for the `:header` (elements that are now in the AppHeader). I would be tempted to audit the current usage, and if we don't find them in use to simplify the API.

Reply:
Alright, I'm thinking to perhaps do this in the follow on. Would that be ok? I'm making a list of smaller optimizations/improvements like this which I could complete as a fast follow on.

Reply:
this is the one bit from the open conversations that defines the component API. technically, moving _away from named blocks_ or _to named blocks_ would both mean a breaking change. I'm ok if we want to do it 'behind the scenes' (assuming consumers will not adopt the component before we decide on the API) but I want it to be a conscious decision and not an afterthought

Reply:
It's looking like everything that was in the :header and the :footer should go in the AppHeader so I'll just remove those named blocks from the AppSideNav

Reply:
@alex-ju I went ahead and removed the :header and :footer blocks, updated related tests, & the Showcase. I'm wondering now if the :body block still makes sense. There is a :root and a :body which the original SideNav had though: https://helios.hashicorp.design/components/side-nav?tab=code#sidenavbase

Reply:
indeed. considering the current implementation, I'd be tempted to only have the default block (no named blocks) to keep things simple. and if we find additional needs we can decide how we want to go about them

Reply:
@alex-ju I ended up reverting removal of the named blocks because it was just becoming over-complicated to refactor all the underlying code. I think it should be done as a follow on if we do it so that it can be reviewed properly in more isolation from all the other pre-existing code.

Reply:
Update: This is being worked on in a separate branch

https://github.com/hashicorp/design-system/pull/2477